extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnSecurity.java

```diff
@@ -1,84 +1,418 @@
+            "DgYDVR0PAQH/BAQDAgEGMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFI/wS3+o" +
+            "LkUkrk1Q+mOai97i3Ru8MA0GCSqGSIb3DQEBCwUAA4IBAQBLQNvAUKr+yAzv95ZU" +
+            "RUm7lgAJQayzE4aGKAczymvmdLm6AC2upArT9fHxD4q/c2dKg8dEe3jgr25sbwMp" +
+            "jjM5RcOO5LlXbKr8EpbsU8Yt5CRsuZRj+9xTaGdWPoO4zzUhw8lo/s7awlOqzJCK" +
+            "6fBdRoyV3XpYKBovHd7NADdBj+1EbddTKJd+82cEHhXXipa0095MJ6RMG3NzdvQX" +
+            "mcIfeg7jLQitChws/zyrVQ4PkX4268NXSb7hLi18YIvDQVETI53O9zJrlAGomecs" +
+            "Mx86OyXShkDOOyyGeMlhLxS67ttVb9+E7gUJTb0o2HLO02JQZR7rkpeDMdmztcpH" +
+            "WD9f";
+
+    @Inject
+    TlsConfigurationRegistry certificates;
 
     @Inject
     WebAuthnAuthenticationMechanism authMech;
+
+    @Inject
+    WebAuthnAuthenticatorStorage storage;
+
+    private ObjectConverter objectConverter = new ObjectConverter();
+    private WebAuthnAsyncManager webAuthn;
+    private VertxContextPRNG random;
+
     private String challengeCookie;
     private String challengeUsernameCookie;
 
+    private String origin;
+    private String rpId;
+    private String rpName;
+
+    private UserVerification userVerification;
+    private Boolean userPresenceRequired;
+    private List<PublicKeyCredentialParameters> pubKeyCredParams;
+    private ResidentKey residentKey;
+
+    private Duration timeout;
+    private int challengeLength;
+    private AuthenticatorAttachment authenticatorAttachment;
+
+    private Attestation attestation;
+
     public WebAuthnSecurity(WebAuthnRunTimeConfig config, Vertx vertx, WebAuthnAuthenticatorStorage database) {
-        // create the webauthn security object
-        WebAuthnOptions options = new WebAuthnOptions();
-        RelyingParty relyingParty = new RelyingParty();
-        if (config.relyingParty().id().isPresent()) {
-            relyingParty.setId(config.relyingParty().id().get());
-        }
-        // this is required
-        relyingParty.setName(config.relyingParty().name());
-        options.setRelyingParty(relyingParty);
-        if (config.attestation().isPresent()) {
-            options.setAttestation(config.attestation().get());
-        }
-        if (config.authenticatorAttachment().isPresent()) {
-            options.setAuthenticatorAttachment(config.authenticatorAttachment().get());
-        }
-        if (config.challengeLength().isPresent()) {
-            options.setChallengeLength(config.challengeLength().getAsInt());
-        }
+        // apply config defaults
+        this.rpId = config.relyingParty().id().orElse(null);
+        this.rpName = config.relyingParty().name();
+        this.origin = config.origin().orElse(null);
+        this.challengeCookie = config.challengeCookieName();
+        this.challengeUsernameCookie = config.challengeUsernameCookieName();
+        this.challengeLength = config.challengeLength().orElse(64);
+        this.userPresenceRequired = config.userPresenceRequired().orElse(false);
```

Comment:
The default value of `userPresenceRequired` should be `true`.
The specification says "If options.[mediation](https://w3c.github.io/webappsec-credential-management/#dom-credentialcreationoptions-mediation) is not set to [conditional](https://w3c.github.io/webappsec-credential-management/#dom-credentialmediationrequirement-conditional), verify that the [UP](https://w3c.github.io/webauthn/#authdata-flags-up) bit of the [flags](https://w3c.github.io/webauthn/#authdata-flags) in authData is set."
https://w3c.github.io/webauthn/#sctn-registering-a-new-credential

Reply:
To automatically replace a password with a passkey when an end-user has already registered a password in their password manager, a new rule has been added to the WebAuthn spec.
when `navigator.credentials.create` is called with `options.mediation` set to `conditional`, a passkey will be automatically created and the `UP` flag will be set to `false` for that passkey.
Apart from this special case, the `UP` flag is expected to always be `true`.

Related Apple WWDC video:
https://developer.apple.com/videos/play/wwdc2024/10125/
WebAuthn spec change:
https://github.com/w3c/webauthn/pull/1951